c/planarityTestGraphFunctionality.c

```diff
@@ -165,7 +165,10 @@ int TestGraphFunctionality(char *commandString, char *infileName, char *inputStr
 						free(inputStr);
 						inputStr = NULL;
 					}
-
+					
+					// Take ownership of strOrFile->theStr, which may have a new address due to having been realloc'ed
```

Comment:
I don't think strOrFile owns the string, nor the FILE if that is what it manages. I think you're just reacquiring the latest pointer value because the address may have changed due to a realloc()